pilot/pkg/serviceregistry/util/label/label.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package label
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/api/label"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/cluster"
+	"istio.io/istio/pkg/config/labels"
+	"istio.io/istio/pkg/network"
+)
+
+// AugmentLabels adds additional labels to the those provided.
+func AugmentLabels(in labels.Instance, clusterID cluster.ID, locality string, networkID network.ID) labels.Instance {
+	// Copy the original labels to a new map.
+	out := make(labels.Instance, len(in)+5)
+	for k, v := range in {
+		out[k] = v
+	}
+
+	// Don't need to add label.TopologyNetwork.Name, since that's already added by injection.
```

Comment:
Confused what we mean by this comment? Seems like we do add it on L50.

Reply:
This is copied from previous implement, i will address separately